lib/RVSDG/RVSDGOps.cpp

```diff
@@ -158,56 +169,66 @@ LogicalResult LambdaNode::verify() {
  *  - Given a function signature with n results of types T1, T2, ..., Tn, the lambda node
  *    terminator should have n operands of types T1, T2, ..., Tn.
  */
-LogicalResult LambdaResult::verify() {
-  auto parent = dyn_cast<LambdaNode>((*this)->getParentOp());
-  if (parent == NULL) {
-    return emitOpError(
-        "LambdaResult has no parent of type LambdaNode. This error should "
-        "never appear, so if it does, may God have mercy on your soul");
-  }
-  Value signatureVal = parent.getResult();
-  if (!signatureVal.getType().isa<LambdaRefType>()) {
-    return emitOpError(
-        "Result type is invalid. Expected an instance of LambdaRefType.");
-  }
-  LambdaRefType signatureType = signatureVal.getType().cast<LambdaRefType>();
-  ArrayRef<Type> signatureReturnTypes = signatureType.getReturnTypes();
-
-  auto resultTypes = this->getOperandTypes();
-
-  if (signatureReturnTypes.size() != resultTypes.size()) {
-    return emitOpError("Number of operands to lambda terminator does not match "
-                       "number of return types in signature.");
-  }
-
-  size_t typeIndex = 0;
-  for (auto [sigType, resType] : zip(signatureReturnTypes, resultTypes)) {
-    if (sigType != resType) {
-      return emitOpError("Type mismatch between lambda signature and lambda "
-                         "result Op. Offending type: #")
-             << typeIndex << " Signature has type " << sigType
-             << ", result Op has type " << resType;
-    }
-    ++typeIndex;
-  }
+LogicalResult
+LambdaResult::verify()
+{
+  // auto parent = dyn_cast<LambdaNode>((*this)->getParentOp());
+  // if (parent == NULL) {
+  //   return emitOpError(
+  //       "LambdaResult has no parent of type LambdaNode. This error should "
+  //       "never appear, so if it does, may God have mercy on your soul");
+  // }
+  // Value signatureVal = parent.getResult();
+  // if (!signatureVal.getType().isa<LambdaRefType>()) {
+  //   return emitOpError(
+  //       "Result type is invalid. Expected an instance of LambdaRefType.");
+  // }
+  // LambdaRefType signatureType = signatureVal.getType().cast<LambdaRefType>();
+  // ArrayRef<Type> signatureReturnTypes = signatureType.getReturnTypes();
+
+  // auto resultTypes = this->getOperandTypes();
+
+  // if (signatureReturnTypes.size() != resultTypes.size()) {
+  //   return emitOpError("Number of operands to lambda terminator does not match "
+  //                      "number of return types in signature.");
+  // }
+
+  // size_t typeIndex = 0;
+  // for (auto [sigType, resType] : zip(signatureReturnTypes, resultTypes)) {
+  //   if (sigType != resType) {
+  //     return emitOpError("Type mismatch between lambda signature and lambda "
+  //                        "result Op. Offending type: #")
+  //            << typeIndex << " Signature has type " << sigType
+  //            << ", result Op has type " << resType;
+  //   }
+  //   ++typeIndex;
+  // }
```

Comment:
Why is validation of LambdaResult commented out?

Reply:
same as above
Maybe checking the parent op is ok, but it has         HasParent<"mlir::rvsdg::LambdaNode">, as an attribute, so I think it is redundant